builtin/logical/ad/path_config.go

```diff
@@ -0,0 +1,219 @@
+package ad
+
+import (
+	"context"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/helper/activedirectory"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+const (
+	// This length is arbitrarily chosen but should work for
+	// most Active Directory minimum and maximum length settings.
+	// A bit of tongue-in-cheek since programmers love their base-2 exponents.
+	defaultPasswordLength = 64
+
+	// The number of minutes in 32 days.
+	defaultPasswordTTLs = 24 * 60 * 32
+)
+
+type configHandler struct {
+	logger hclog.Logger
+}
+
+func (h *configHandler) Path() *framework.Path {
+	return &framework.Path{
+		Pattern: "config",
+		Fields: map[string]*framework.FieldSchema{
+			"username": {
+				Type:        framework.TypeString,
+				Default:     "",
+				Description: "Username with sufficient permissions in Active Directory to administer passwords.",
+			},
+
+			"password": {
+				Type:        framework.TypeString,
+				Default:     "",
+				Description: "Password for username with sufficient permissions in Active Directory to administer passwords.",
+			},
+
+			"url": {
```

Comment:
Can we rename this to `urls` to make the intent explicit?

Reply:
I like that and I think it would be clearer. However, there's a precedent on other LDAP backends to just have it as `"url"`, ex. https://github.com/hashicorp/vault/blob/master/builtin/credential/ldap/path_config.go#L27. I'm not sure if it's better to go with the precedent, or to go with something more descriptive. Thoughts?